upmc-enterprises/cda2fhir
# Feature/upmcfhir 218 medication supply order

#### What does this PR do?

Adds MedicationSupplyOrder -> MedicationRequest transformation function to the ResourceTransfromer. Also adds generators for MedicationSupplyOrder and ManufacturedProduct (a child element of MedicationSupplyOrder). Refactors some code in used by both MedicationDispense, MedicationStatement, and MedicationRequest for generating fhir medications from cda manufactured products. Removes the method for MedicationInformation -> medication because it was just calling ManufacutedProduct->Medication.

#### Related JIRA tickets:
[UPMCFHIR-218](https://jira.amida.com/browse/UPMCFHIR-218)
#### How should this be manually tested?
Run any of the test data through, and check that any <supply classCode="SPLY" moodCode="INT"> elements are turned into MedicationRequests, with their Medication's referenced. 

#### Background/Context:

#### New JIRA tickets for clinical review required?
Yes. [UPMCFHIR-279](https://jira.amida.com/browse/UPMCFHIR-279)
